planemo/galaxy_config.py

```diff
@@ -354,7 +355,8 @@ def _ensure_galaxy_repository_available(ctx):
         shell("git --git-dir %s fetch >/dev/null 2>&1" % gx_repo)
     else:
         remote_repo = "https://github.com/galaxyproject/galaxy"
-        shell("git clone --bare %s %s" % (remote_repo, gx_repo))
+        command = git.command_clone(ctx, remote_repo, gx_repo, bare=True)
+        shell(command)
```

Comment:
So this line is pulling Galaxy down for ``test``/``serve`` when used with ``--install_galaxy`` - which used to use a tar ball by default  but now does a clone so that it can do incremental updates on each call very quickly.

It sounds like you mean for the actual shed stuff though - which could be optimized somewhat. I don't think we should just download the tars though because I would like to use the git metadata during upload (#170) and because this will need to support submodules at some point. I do think we could do ``clone`` with ``--depth=1`` probably to speed things up though paired with creating a directory to cache repositories between calls (mirroring how Galaxy is quickly cloned).